<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <author>Marcelo Tellier Sartori Vaz</author>
 <class>mainWindow</class>
 <widget class="QMainWindow" name="mainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>NBR 5410 Calculator</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0" colspan="4">
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="projectTab">
       <attribute name="title">
        <string>Project</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="2">
         <widget class="QPushButton" name="newSupplyButton">
          <property name="icon">
           <iconset theme="contact-new"/>
          </property>
         </widget>
        </item>
        <item row="0" column="8">
         <widget class="QPushButton" name="newWireTypeButton">
          <property name="icon">
           <iconset theme="contact-new"/>
          </property>
         </widget>
        </item>
        <item row="0" column="4">
         <spacer name="horizontalSpacer2">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="3">
         <widget class="QLabel" name="loadTypesLabel">
          <property name="text">
           <string>Load Types</string>
          </property>
         </widget>
        </item>
        <item row="0" column="6">
         <widget class="QLabel" name="wireTypesLabel">
          <property name="text">
           <string>Wire Types</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3" colspan="3">
         <widget class="LoadTypeView" name="loadTypesListView"/>
        </item>
        <item row="0" column="0">
         <widget class="QLabel" name="suppliesLabel">
          <property name="text">
           <string>Supplies</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="3">
         <widget class="SupplyView" name="suppliesListView"/>
        </item>
        <item row="0" column="1">
         <spacer name="horizontalSpacer1">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="5">
         <widget class="QPushButton" name="newLoadTypeButton">
          <property name="icon">
           <iconset theme="contact-new"/>
          </property>
         </widget>
        </item>
        <item row="1" column="6" colspan="3">
         <widget class="WireTypeView" name="wireTypesListView"/>
        </item>
        <item row="0" column="7">
         <spacer name="horizontalSpacer3">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="circuitsTab">
       <attribute name="title">
        <string>Circuits</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="1" column="0" colspan="4">
         <widget class="CircuitsView" name="circuitsTreeView"/>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="deleteCircuitButton">
          <property name="text">
           <string>Delete circuit</string>
          </property>
          <property name="icon">
           <iconset theme="delete"/>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <spacer name="horizontalSpacer4">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="0" column="3">
         <widget class="QPushButton" name="newCircuitButton">
          <property name="text">
           <string>New circuit</string>
          </property>
          <property name="icon">
           <iconset theme="window-new"/>
          </property>
         </widget>
        </item>
        <item row="0" column="2">
         <widget class="QPushButton" name="newUpstreamCircuitButton">
          <property name="text">
           <string>New upstream circuit</string>
          </property>
          <property name="icon">
           <iconset theme="window-new"/>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="conduitsTab">
       <attribute name="title">
        <string>Conduits</string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="0" column="2">
         <widget class="QPushButton" name="newConduitRunButton">
          <property name="text">
           <string>New conduit run</string>
          </property>
          <property name="icon">
           <iconset theme="window-new"/>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QPushButton" name="deleteConduitRunButton">
          <property name="text">
           <string>Delete conduit run</string>
          </property>
          <property name="icon">
           <iconset theme="delete"/>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <spacer name="horizontalSpacer5">
          <property name="orientation">
           <enum>Qt::Orientation::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
        <item row="1" column="0" colspan="3">
         <widget class="ConduitRunsView" name="conduitsTreeView"/>
        </item>
        <item row="2" column="0" colspan="3">
         <widget class="UnassignedCircuitsView" name="unassignedCircuitsView"/>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1000</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>&amp;File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="separator"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionQuit"/>
   </widget>
   <widget class="QMenu" name="menuHelp">
    <property name="title">
     <string>&amp;Help</string>
    </property>
    <addaction name="actionAbout"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuHelp"/>
  </widget>
  <action name="actionNew">
   <property name="icon">
    <iconset theme="document-new"/>
   </property>
   <property name="text">
    <string>&amp;New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="icon">
    <iconset theme="document-open"/>
   </property>
   <property name="text">
    <string>&amp;Open...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="icon">
    <iconset theme="document-save"/>
   </property>
   <property name="text">
    <string>&amp;Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="icon">
    <iconset theme="document-save-as"/>
   </property>
   <property name="text">
    <string>Save &amp;As...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+S</string>
   </property>
  </action>
  <action name="actionQuit">
   <property name="icon">
    <iconset theme="application-exit"/>
   </property>
   <property name="text">
    <string>&amp;Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
  <action name="actionAbout">
   <property name="icon">
    <iconset theme="help-about"/>
   </property>
   <property name="text">
    <string>&amp;About</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SupplyView</class>
   <extends>QListView</extends>
   <header>nbr_5410_calculator.projectTab</header>
   <slots>
    <slot>deleteSelectedItems()</slot>
    <slot>newSupply()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>LoadTypeView</class>
   <extends>QListView</extends>
   <header>nbr_5410_calculator.projectTab</header>
   <slots>
    <slot>deleteSelectedItems()</slot>
    <slot>newLoadType()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>WireTypeView</class>
   <extends>QListView</extends>
   <header>nbr_5410_calculator.projectTab</header>
   <slots>
    <slot>deleteSelectedItems()</slot>
    <slot>newWireType()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>CircuitsView</class>
   <extends>QTreeView</extends>
   <header>nbr_5410_calculator.circuitsTab</header>
   <slots>
    <slot>newCircuit()</slot>
    <slot>deleteSelectedItems()</slot>
    <slot>newUpstreamCircuit()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>ConduitRunsView</class>
   <extends>QTreeView</extends>
   <header>nbr_5410_calculator.conduitsTab</header>
   <slots>
    <slot>newConduitRun()</slot>
    <slot>deleteSelectedItems()</slot>
   </slots>
  </customwidget>
  <customwidget>
   <class>UnassignedCircuitsView</class>
   <extends>QListView</extends>
   <header>nbr_5410_calculator.conduitsTab</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionSave</sender>
   <signal>triggered()</signal>
   <receiver>mainWindow</receiver>
   <slot>saveProject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>triggered()</signal>
   <receiver>mainWindow</receiver>
   <slot>loadProject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSaveAs</sender>
   <signal>triggered()</signal>
   <receiver>mainWindow</receiver>
   <slot>saveProject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newCircuitButton</sender>
   <signal>clicked()</signal>
   <receiver>circuitsTreeView</receiver>
   <slot>newCircuit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>968</x>
     <y>90</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>337</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionNew</sender>
   <signal>triggered()</signal>
   <receiver>mainWindow</receiver>
   <slot>newProject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionAbout</sender>
   <signal>triggered()</signal>
   <receiver>mainWindow</receiver>
   <slot>showAbout()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionQuit</sender>
   <signal>triggered()</signal>
   <receiver>mainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>299</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>deleteCircuitButton</sender>
   <signal>clicked()</signal>
   <receiver>circuitsTreeView</receiver>
   <slot>deleteSelectedItems()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>726</x>
     <y>93</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>337</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newConduitRunButton</sender>
   <signal>clicked()</signal>
   <receiver>conduitsTreeView</receiver>
   <slot>newConduitRun()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>914</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>337</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>deleteConduitRunButton</sender>
   <signal>clicked()</signal>
   <receiver>conduitsTreeView</receiver>
   <slot>deleteSelectedItems()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>764</x>
     <y>91</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>337</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newUpstreamCircuitButton</sender>
   <signal>clicked()</signal>
   <receiver>circuitsTreeView</receiver>
   <slot>newUpstreamCircuit()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>807</x>
     <y>81</y>
    </hint>
    <hint type="destinationlabel">
     <x>499</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newSupplyButton</sender>
   <signal>clicked()</signal>
   <receiver>suppliesListView</receiver>
   <slot>newSupply()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>311</x>
     <y>80</y>
    </hint>
    <hint type="destinationlabel">
     <x>172</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newLoadTypeButton</sender>
   <signal>clicked()</signal>
   <receiver>loadTypesListView</receiver>
   <slot>newLoadType()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>639</x>
     <y>80</y>
    </hint>
    <hint type="destinationlabel">
     <x>492</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>newWireTypeButton</sender>
   <signal>clicked()</signal>
   <receiver>wireTypesListView</receiver>
   <slot>newWireType()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>965</x>
     <y>80</y>
    </hint>
    <hint type="destinationlabel">
     <x>819</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>saveProject()</slot>
  <slot>loadProject()</slot>
  <slot>newProject()</slot>
  <slot>showAbout()</slot>
 </slots>
</ui>
